core/src/test/java/org/apache/iceberg/TestScansAndSchemaEvolution.java

```diff
@@ -113,4 +118,48 @@ public void testPartitionSourceRename() throws IOException {
 
     assertThat(tasks).hasSize(1);
   }
+
+  @TestTemplate
+  public void testColumnRenameOrDrop() throws IOException {
+    File location = Files.createTempDirectory(temp, "junit").toFile();
+    assertThat(location.delete()).isTrue(); // should be created by table create
+
+    Table table = TestTables.create(location, "test", SCHEMA, SPEC, formatVersion);
+
+    DataFile fileOne = createDataFile("one");
+    DataFile fileTwo = createDataFile("two");
+
+    table.newAppend().appendFile(fileOne).appendFile(fileTwo).commit();
+    long firstSnapshotId = table.currentSnapshot().snapshotId();
+
+    table.updateSchema().renameColumn("data", "renamed_data").commit();
```

Comment:
How about testing time travel after partition evolution too?

Reply:
Why is this comment closed?
I think @ebyhr suggestion is actually a valid one

Reply:
also added a drop case for partition